### PR TITLE
Add new package: sbml

### PR DESCRIPTION
**Pull request by rblake-llnl**
Libraries and bindings for the Systems Biology Markup Language.

I only added python support variants, but additional support for other languages could also be added.

### CODE REVIEW

**Comment by rblake-llnl** [comment]
No idea why the build is failing-- it seems to be related to tty echoing, which my package doesn't touch.  Are the unit tests bugged?

File: var/spack/repos/builtin/packages/sbml/package.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        if '+python' in spec:
+            args.extend([
+                "-DWITH_PYTHON:BOOL=ON",
+                "-DWITH_PYTHON_INCLUDE:PATH=%s" % spec['python'].prefix,
+            ])
+
+        if '+mono' in spec:
+            args.append("-DWITH_CSHARP:BOOL=ON")
+        else:
+            args.append("-DWITH_CSHARP:BOOL=OFF")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/sbml/package.py]: I don't see the `mono` variant in the declarations above. Can we either add it as a variant or add a TODO here that clearly points out that the condition at the moment will always be False?

File: var/spack/repos/builtin/packages/sbml/package.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+        if '+java' in spec:
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/sbml/package.py:99]: Similarly for `java`, `matlab`, `octave`, `r` and `ruby` variants.

File: var/spack/repos/builtin/packages/sbml/package.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+class Sbml(CMakePackage):
+    """Library for the Systems Biology Markup Language"""
+
+    maintainers = ['rblake-llnl']
+    homepage = "https://sbml.org"
+
+    def url_for_version(self, version):
+        url = "https://downloads.sourceforge.net/project/sbml/libsbml/{0}/stable/libSBML-{1}-core-plus-packages-src.tar.gz".format(version, version)
+        return url
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/sbml/package.py]: Minor OCD: can we move this method below the directives? Asking as that will be consistent with most packages already in the builtin repository.

**Review by alalazo — CHANGES_REQUESTED** [review]
This basically LGTM, the only request for change is on variants. Can we either add them explicitly or put TODO comments where non-existing variants are checked in the recipe?

**Comment by alalazo** [comment]
> Are the unit tests bugged?

That single unit test, as far as I know, is non-deterministic - so sometimes it fails spuriously. I just restarted the MacOS tests and am quite confident they'll succeed this time.

File: var/spack/repos/builtin/packages/sbml/package.py (hunk 3 above)
Comment (rblake-llnl) [reply var/spack/repos/builtin/packages/sbml/package.py]: Done!

File: var/spack/repos/builtin/packages/sbml/package.py (hunk 1 above)
Comment (rblake-llnl) [reply var/spack/repos/builtin/packages/sbml/package.py]: Adding support for all those languages, not just the ones I needed.

File: var/spack/repos/builtin/packages/sbml/package.py [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+                "-DWITH_PYTHON:BOOL=ON",
+                "-DWITH_PYTHON_INCLUDE:PATH=%s" % spec['python'].prefix,
+            ])
+        else:
+            args.append('-DWITH_PYTHON:BOOL=ON')
+
+        if '+mono' in spec:
+            args.append("-DWITH_CSHARP:BOOL=ON")
+        else:
+            args.append("-DWITH_CSHARP:BOOL=OFF")
```
Comment (becker33) [inline var/spack/repos/builtin/packages/sbml/package.py]:
```suggestion
        args.append(self.define_from_variant('WITH_CSHARP', 'mono'))
```

File: var/spack/repos/builtin/packages/sbml/package.py [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+
+        if "+r" in spec:
+            args.append("-DWITH_R:BOOL=ON")
+        else:
+            args.append("-DWITH_R:BOOL=OFF")
+
+        if "+ruby" in spec:
+            args.append("-DWITH_RUBY:BOOL=ON")
+        else:
+            args.append("-DWITH_RUBY:BOOL=OFF")
```
Comment (becker33) [inline var/spack/repos/builtin/packages/sbml/package.py]:
```suggestion
        args.append(self.define_from_variant('WITH_OCTAVE', 'octave'))
        args.append(self.define_from_variant('WITH_PERL', 'perl'))
        args.append(self.define_from_variant('WITH_R', 'r'))
        args.append(self.define_from_variant('WITH_RUBY', 'ruby'))
```

**Review by becker33 — COMMENTED** [review]
@rblake-llnl we have a nice helper method for defining CMake variables from variants, this could clean the code up a fair bit (for the ones that don't require special handling).

**Comment by rblake-llnl** [comment]
Thanks for the tips, @becker33 !  I merged your changes.

**Review by becker33 — APPROVED** [review]

**Pull request merged**